qiskit/ignis/verification/randomized_benchmarking/circuits.py

```diff
@@ -364,24 +401,50 @@ def randomized_benchmarking_seq(nseeds=1, length_vector=None,
                     # add measurement for interleaved rb
                     circ_interleaved.measure(qr[qb], cr[qind])
 
-                circ.name = 'rb_length_%d_seed_%d' % (length_index,
-                                                      seed + seed_offset)
-                circ_interleaved.name = 'rb_interleaved_length_%d_seed_%d' \
-                                        % (length_index, seed + seed_offset)
+                circ.name = \
+                    rb_circ_type + '_length_%d_seed_%d' % \
+                    (length_index, seed + seed_offset)
+                circ_interleaved.name = \
+                    rb_circ_type + '_interleaved_length_%d_seed_%d' % \
+                    (length_index, seed + seed_offset)
+
+                if group_gates == 'Non-Clifford':
+                    circ.name = rb_circ_type + '_Z_length_%d_seed_%d' % \
+                                (length_index, seed + seed_offset)
+                    circ_interleaved.name = \
+                        rb_circ_type + '_interleaved_Z_length_%d_seed_%d' % \
+                        (length_index, seed + seed_offset)
+                    nonclifford_circ.name = \
+                        rb_circ_type + '_X_length_%d_seed_%d' % \
+                        (length_index, seed + seed_offset)
+                    nonclifford_interleaved_circ.name = \
+                        rb_circ_type + 'interleaved_X_length_%d_seed_%d' % \
+                        (length_index, seed + seed_offset)
 
                 circuits[seed].append(circ)
                 circuits_interleaved[seed].append(circ_interleaved)
+                circuits_nonclifford[seed].append(nonclifford_circ)
+                circuits_nonclifford_interleaved[seed].append(
+                    nonclifford_interleaved_circ)
+
                 if is_purity:
                     for d in range(npurity):
                         circuits_purity[seed][d].append(circ_purity[d])
                 length_index += 1
 
-    # output of interleaved rb
-    if interleaved_gates is not None:
-        return circuits, xdata, circuits_interleaved
     # output of purity rb
     if is_purity:
         return circuits_purity, xdata, npurity
+    # output of non-clifford interleaved rb
+    if interleaved_gates is not None and group_gates == 'Non-Clifford':
+        return circuits, xdata, circuits_nonclifford, circuits_interleaved, \
+               circuits_nonclifford_interleaved
+    # output of interleaved rb
+    if interleaved_gates is not None:
+        return circuits, xdata, circuits_interleaved
+    # output of Non-Clifford rb
+    if group_gates == 'Non-Clifford':
+        return circuits, xdata, circuits_nonclifford
```

Comment:
I would still return `circuits, xdata`, except `circuits` is a 2xn list

Reply:
I used the similar convention to the one that was used for interleaved RB, since in this case we have two fitters (and two graphs), unlike purity RB where there is only one fitter and graph.

Reply:
(so I didn't make any changes here)

Reply:
Ok, leave it for now, but I would consider changing it across the board later (ie for interleaved too) because the ordering seems confusing